[HIVEMALL-5-pre] Java code formatter

takuti opened the pull request:
## What changes were proposed in this pull request?

In order to make contribution to Hivemall more easier, we need to prepare contribution guide as soon as possible. This PR is the first step toward publishing a complete document on https://hivemall.incubator.apache.org

For now, I added (temporal) contribution guide to README and require contributors to follow specific cording conventions hereafter.

## What type of PR is it?

Documentation?

## What is the Jira issue?

https://issues.apache.org/jira/browse/HIVEMALL-5

---

Originally, `eclipse-style.xml` was created based on google's coding conventions. However, in this PR, I suggest to follow coding conventions of Hive hereafter; since Hivemall clearly stands on Hive, following its conventions should be the most reasonable choice.

@myui If you strongly want to use different kind of conventions, I will update this PR so.

myui commented on core/src/main/java/hivemall/anomaly/SDAR2D.java:
```diff
@@ -33,129 +33,123 @@
+    // solve A in the following Yule-Walker equation
+    // C_j = ∑_{i=1}^{k} A_i C_{j-i} where j = 1..k, C_{-i} = C_i'
+    /*
+     * /C_1\ /A_1\ /C_0 |C_1' |C_2' | . . . |C_{k-1}' \ |---| |---| |--------+--------+--------+
```
This comment should not be formatted.

myui commented on core/src/main/java/hivemall/classifier/AROWClassifierUDTF.java:
```diff
@@ -41,172 +41,165 @@
-                + " - Returns a relation consists of <string|int|bigint feature, float weight, float covar>",
-        extended = "Build a prediction model by Adaptive Regularization of Weight Vectors (AROW) binary classifier")
+@Description(name = "train_arow", value = "_FUNC_(list<string|int|bigint> features, int label [, const string options])"
+    + " - Returns a relation consists of <string|int|bigint feature, float weight, float covar>", extended = "Build a prediction model by Adaptive Regularization of Weight Vectors (AROW) binary classifier")
```
too long line.

myui commented on core/src/main/java/hivemall/anomaly/SingularSpectrumTransform.java:
```diff
@@ -41,161 +41,160 @@
+      }
+
+      switch (scoreFunc) {
+      case svd:
```
`case` better to be indented as seen in Google coding style.

myui commented on core/src/main/java/hivemall/classifier/KPAPredictUDAF.java:
```diff
@@ -40,189 +40,187 @@
 
-            return PrimitiveObjectInspectorFactory.writableDoubleObjectInspector;
-        }
+    public Evaluator() {
```
`public Evaluator() {}` is preferred as seen in Google coding style.
https://google.github.io/styleguide/javaguide.html#s4.1.3-braces-empty-blocks

myui commented on core/src/main/java/hivemall/utils/lang/Preconditions.java:
```diff
@@ -26,16 +26,17 @@
 public final class Preconditions {
 
-    private Preconditions() {}
+  private Preconditions() {
```
`private Preconditions() {}` is preferred.

myui commented:
I need to rewrite [JIRA issue](https://issues.apache.org/jira/browse/HIVEMALL-5). Let's discuss on the next Monday (or Tues).

takuti commented:
@myui Could you share how you are planning to re-write the JIRA issue here If possible? Keeping our discussion in public is generally better :)

Overall, I suspect that you want to follow the google coding convention rather than Hive's one. Is it correct? I'm personally okay for any conventions as long as they are well-documented. If we define some exceptions on either google / Hive (oracle) conventions, we manually need to document them ✍️

myui commented:
>  I suspect that you want to follow the google coding convention rather than Hive's one. 

Right.

But, completely following Google coding style has problems in formatting existing codes. Manual comment rewriting and string concatenation rewriting is required.

myui commented on core/src/main/java/hivemall/anomaly/SDAR2D.java:
```diff
@@ -33,129 +33,123 @@
+  public RealVector update(@Nonnull final ArrayRealVector[] x, final int k) {
+    Preconditions.checkArgument(x.length >= 1, "x.length MUST be greather than 1: " + x.length);
+    Preconditions.checkArgument(k >= 0, "k MUST be greather than or equals to 0: ", k);
+    Preconditions.checkArgument(k < _C.length, "k MUST be less than |C| but " + "k=" + k + ", |C|="
```
Here, you can find String concatenation problems.

takuti replied:
@myui Oh, it's huge...I understand what you are afraid ;(

myui commented:
Current `eclipse-style.xml` is using modified version of Sun's coding style (no comment formatting), not google one. It's better to evaluate `Google coding style + no comment formatting` though. 

Google coding style provides both Eclipse format and IntelliJ format but different formatting is applied due to mismatch in functionality of IDE. So, providing only Eclipse format (which can be loaded in IntelliJ) is preferred.

takuti commented:
@myui Thank you for sharing your opinion! I realized that this issue requires us to undergo significant amount of discussion and modification of the current code.

So, I renamed the title of this PR to **[HIVEMALL-5-pre]** as the first step to publish our fixed version of coding guideline. There is no update in terms of coding conventions, and I just applied the formatter to the current code and updated README for future contributors.

If it's acceptable for you, I like to merge this soon :)

myui commented on core/src/main/java/hivemall/anomaly/SingularSpectrumTransform.java:
```diff
@@ -165,8 +165,8 @@ private double computeScoreSVD(@Nonnull final RealMatrix H, @Nonnull final RealM
      *
-     * Number of iterations for the Power method and QR method is fixed to 1 for efficiency. This
-     * may cause failure (i.e. meaningless scores) depending on datasets and initial values.
+     * Number of iterations for the Power method and QR method is fixed to 1 for efficiency. This may cause failure (i.e. meaningless scores)
```
Better to set Javadoc comment formatting off for the time being.

myui commented on core/src/main/java/hivemall/ftvec/binning/NumericHistogram.java:
```diff
@@ -30,11 +30,9 @@
- * data and a large (e.g., 20-80) number of histogram bins.
+ * A generic, re-usable histogram class that supports partial aggregations. The algorithm is a heuristic adapted from the following paper: Yael
+ * Ben-Haim and Elad Tom-Tov, "A streaming parallel decision tree algorithm", J. Machine Learning Research 11 (2010), pp. 849--872. Although there are
+ * no approximation guarantees, it appears to work well with adequate data and a large (e.g., 20-80) number of histogram bins.
```
same as here.

myui commented:
Merged with some modifications. 100 spaces for Javadoc comment, as same as Google style, works fine.
